### PR TITLE
Make passing build args explicit in ci/prod builds

### PR DESCRIPTION
**Pull request by potiuk**
When building hte image, breeze converts some simple parameters passed as breeze command (with autocompletion and explanation) into much longer and more complex set of build args that are passed to `docker build` command. The way how passing hte args worked so far is that it was pretty implicit:

* **kwargs were used to ingest `click` flags
* parameters found as empty/None were filtered out from these
* Build*Params dataclass was created out of such kwargs dict
* argumenst from dataclass (with some customization) were converted to --build-arg (CAPITALIZED_PROPERTY_NAME)

This had a lot of implicitness and it was not easy to understand whether the parameters passed were correct and how they passed through this chain.

This change makes all the build arg much more explicit - without kwargs and dictionary. Each CI/PROD build param has now a method where it explicitly converts arguments into build-args - including specifying which of those are optional (where you can actually filter out Empty and None values) and which are required (where an actual value is expected).

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/commands/ci_image_commands.py [hunk 1]
```diff
@@ -550,13 +614,6 @@ def run_build_ci_image(
     :param ci_image_params: CI image parameters
     :param output: output redirection
     """
-    if not ci_image_params.version_suffix_for_pypi:
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/ci_image_commands.py:553]: Moved as default value for CI build

File: dev/breeze/src/airflow_breeze/commands/ci_image_commands_config.py [hunk 2]
```diff
@@ -35,7 +35,7 @@
                 "--image-tag",
                 "--tag-as-latest",
                 "--docker-cache",
-                "--force-build",
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/ci_image_commands_config.py:38]: It was not used. When you order a build build is forced.

**Comment by potiuk** [comment]
cc: @Bowrna -> rewritten the old breeze argument  passing now to be more explicit :)

**Comment by potiuk** [comment]
I checked carefully and mapping of parmeters to build args looks good - I also cleaned up some duplicates along the way (like duplicate --progress`

**Review by vincbeck — APPROVED** [review]

**Pull request merged**